### DISCUSSION TITLE
Setup when using a different WEB_DOMAIN to LOCAL_DOMAIN

### DISCUSSION THREAD

**Discussion by cooperaj** [mastodon/mastodon#20667] [Q&A]
I've got my instance at 'social.example.com' but the local domain is 'example.com'. This for the most part appears to work fine as I have added a 301 redirect for the webfinger endpoint to 'example.com'. But what I'm also getting is a lot of requests for '/api/v1/instance' and '/.well-known/host-meta'. 

Should I also be redirecting these? I had noticed that the twitter crossposter was unable to use my user account unless I included the full web domain. Also the user migration tool gave me a number of errors about being unable to find my instance until it just works (probably coincidental).

Thanks

**Comment by oliverlorenz** [comment]
Have you found a solution?

I have the same or at least similar problem. I have a Mastodon instance running on mastodon.myfullname.tdl (I am and remain the only user on this instance). I have set up a redirect on myfullname.tdl. myfullname.tdl is running on firebase, so the config (firebase.json) looks like this:

```
{
  "hosting": {
    "public": "dist",
    "ignore": ["firebase.json"],
    "redirects": [
      [...]
      {
        "source": "/.well-known/host-meta",
        "destination": "https://mastodon.myfullname.tld/.well-known/host-meta?resource=acct:oliver@mastodon.myfullname.tld",
        "type": 301
      },
      {
        "source": "/.well-known/webfinger",
        "destination": "https://mastodon.myfullname.tld/.well-known/webfinger?resource=acct:oliver@mastodon.myfullname.tld",
        "type": 301
      },
      {
        "source": "/.well-known/nodeinfo",
        "destination": "https://mastodon.myfullname.tld/.well-known/nodeinfo?resource=acct:oliver@mastodon.myfullname.tld",
        "type": 301
      }
    ],
    [...]
  }
}

```

When I search for my user on another instance (searchterm "@firstname@myfullname.tdl" I find my account, but it is displayed as "@firstname@mastodon.myfullname.tdl" instead of "@firstname@myfullname.tdl" as I would like.

Can anyone give me a hint what I am doing wrong?

**Reply by cooperaj** [reply]
I think this is because you're misconfiguring the resource that you're redirecting to? The resource should be acct:oliver@myfullname.tld
